automerge-backend/src/actor_states.rs

```diff
@@ -0,0 +1,95 @@
+use crate::error::AutomergeError;
+use crate::protocol::{ActorID, Change, Clock};
+use std::collections::HashMap;
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct ActorStates {
+    change_by_actor: HashMap<ActorID, Vec<Change>>,
+    deps_by_actor: HashMap<ActorID, Vec<Clock>>,
+    pub history: Vec<Change>,
+}
+
+impl ActorStates {
+    pub(crate) fn new() -> ActorStates {
+        ActorStates {
+            change_by_actor: HashMap::new(),
+            deps_by_actor: HashMap::new(),
+            history: Vec::new(),
+        }
+    }
+
+    pub fn is_concurrent(
+        &self,
+        actor_id1: &ActorID,
+        seq1: u32,
+        actor_id2: &ActorID,
+        seq2: u32,
+    ) -> bool {
+        let clock1 = self.get_deps(actor_id1, seq1).unwrap();
```

Comment:
Can we use default empty clocks here instead of panicking? I _think_ the semantics would still be correct. An empty clock will return 0 for all actor IDs.

Reply:
I fussed with this for a bit that appealed to my need to use references instead of copying as much as possible.  Unfortunately it required me to add a blank clock to reference.  Let me know if this passes your sniff test or if there's a cleaner way.